Add documentation for parameters to the now() function (#92)
opendatasoft/ods-documentation-explore-api: source/includes/v2/_odsql.md

A scalar function can be used in [select arithmetic expressions](#arithmetic-select-expression), [filter expressions](#filter-expression) or in the [group by clause](#group-by-clause).

Function|Parameters|Description
--------|-----------|----------
`length`|string literal or string field literal|Returns the number of characters

> `now`|optional named parameters (see below)|Returns the current date

Reviewer: If you include a link to the desired section instead of "see below", it should better help readers find the relevant info.
```suggestion
`now`|optional named [parameters](#parameters-to-the-now-function)|Returns the current date
```

`year`|date field literal|Returns the year of a date
`month`|date field literal|Returns the month of a date
`day`|date field literal|Returns the day of a date
`hour`|date field literal|Returns the hour of a date
`minute`|date field literal|Returns the minute of a date
`second`|date field literal|Returns the second of a date
`date_format`|date field literal|Returns a formatted date

#### Parameters to the now() function

> > Examples, assuming the current date time is 2021-05-06 12:34:55.450500+00:00, which is a thursday

Reviewer: Names of days should be capitalized.
```suggestion
> Examples, assuming the current date time is 2021-05-06 12:34:55.450500+00:00, which is a thursday
```

```sql
now() -- Returns '2021-05-06T12:34:55.450500+00:00'

> now(year=2000) -- Sets the year component to return '2000-05-06T12:34:55.450500+00:00'

Reviewer 2: I know it's well documented with extensive parameters documentation below but it may be misleading to not have an example with relative year here, and it's used a lot(?)
```suggestion
now(year=2000) -- Sets the year component to return '2000-05-06T12:34:55.450500+00:00'
now(years=-1) -- Sets the year to one year ago  which is '2020-05-06T12:34:55.450500+00:00'
```

Author: Thanks. There is just one extra space in your suggestion (or a missing comma maybe)

Reviewer 2: fixed!

> now(year=2001, months=-1) -- Sets the year to 2001 and substract 1 month to return '2000-04-06T12:34:55.450500+00:00'

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
now(year=2001, months=-1) -- Sets the year to 2001 and subtract 1 month to return '2000-04-06T12:34:55.450500+00:00'
```

Author: ! I did not know it was a mistake ! Thanks

now(day=31,month=2) -- Sets the day to 31, then the month to 2. The actual day part is rounded to 28 '2021-02-28T12:34:55.450500+00:00'

> now(weekday=0) -- Sets the day to the next monday which is '2021-05-10T12:34:55.450500+00:00'

Reviewer: Names of days should be capitalized.
```suggestion
now(weekday=0) -- Sets the day to the next Monday which is '2021-05-10T12:34:55.450500+00:00'
```

> now(mondays=+1) -- Sets the day to the next monday which is also '2021-05-10T12:34:55.450500+00:00'

Reviewer: Names of days should be capitalized.
```suggestion
now(mondays=+1) -- Sets the day to the next Monday which is also '2021-05-10T12:34:55.450500+00:00'
```

> now(mondays=-1) -- Sets the day to the previous monday which is '2021-05-03T12:34:55.450500+00:00'

Reviewer: Names of days should be capitalized.
```suggestion
now(mondays=-1) -- Sets the day to the previous Monday which is '2021-05-03T12:34:55.450500+00:00'
```

```

Without any parameters, the `now()` function returns the current date and time.

The function may also be called with named parameters to set or modify certain parts of the current date and time.

> With each parameter an integer value is required, interpreted as an absolute value or as a relative value to a part of the current date and time.

Reviewer: A comma is missing after the introductory clause.
```suggestion
With each parameter, an integer value is required, interpreted as an absolute value or as a relative value to a part of the current date and time.
```

> Parameter names in their singular form will set a certain part of the current date and time to the given value. Parameter names written in plural will add or substract the given value to a part of the current date and time.

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
Parameter names in their singular form will set a certain part of the current date and time to the given value. Parameter names written in plural will add or subtract the given value to a part of the current date and time.
```

If a parameter is used multiple times in the call, only the last one is actually used, the others are ignored.

Parameter name|Accepted values|Description
--------------|---------------|-----------
`year`|1 to 9999|Year component

> `years`|Any integer|Value to add to or substract from the year component

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
`years`|Any integer|Value to add to or subtract from the year component
```

`month`|1 to 12|Month component

> `months`|Any integer|Value to add to or substract from the month component, then the year component in case of overflow

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
`months`|Any integer|Value to add to or subtract from the month component, then the year component in case of overflow
```

`day`|Any positive integer|Day component, rounded to the maximum valid day number for the current month

> `days`|Any integer|Value to add to or substract from the day component, then the month component in case of overflow

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
`days`|Any integer|Value to add to or subtract from the day component, then the month component in case of overflow
```

`hour`|0 to 23|Hour component

> `hours`|Any integer|Value to add to or substract from the hour component, then the day component in case of overflow

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
`hours`|Any integer|Value to add to or subtract from the hour component, then the day component in case of overflow
```

`minute`|0 to 59|Minute component

> `minutes`|Any integer|Value to add to or substract from the minute component, then the hour component in case of overflow

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
`minutes`|Any integer|Value to add to or subtract from the minute component, then the hour component in case of overflow
```

`second`|0 to 59|Second component

> `seconds`|Any integer|Value to add to or substract from the second component, then the minute component in case of overflow

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
`seconds`|Any integer|Value to add to or subtract from the second component, then the minute component in case of overflow
```

`microsecond`|0 to 999999|Microsecond component

> `microseconds`|Any integer|Value to add to or substract from the microsecond component, then the second component in case of overflow

Reviewer: There's a typo here: "substract" --> subtract.
```suggestion
`microseconds`|Any integer|Value to add to or subtract from the microsecond component, then the second component in case of overflow
```

`weekday`|0 to 6|Day of the week, 0 for monday to 6 for sunday

> `mondays`|Any integer|Number of mondays to add to or substract from the current date

Reviewer: - There's a typo here: "substract" --> subtract.
- Names of days should be capitalized.
```suggestion
`mondays`|Any integer|Number of Mondays to add to or subtract from the current date
```

> `tuesdays`|Any integer|Number of tuesdays to add to or substract from the current date

Reviewer: - There's a typo here: "substract" --> subtract.
- Names of days should be capitalized.
```suggestion
`tuesdays`|Any integer|Number of Tuesdays to add to or subtract from the current date
```

> `wednesdays`|Any integer|Number of wednesdays to add to or substract from the current date

Reviewer: - There's a typo here: "substract" --> subtract.
- Names of days should be capitalized.
```suggestion
`wednesdays`|Any integer|Number of Wednesdays to add to or subtract from the current date
```

> `thursdays`|Any integer|Number of thursdays to add to or substract from the current date

Reviewer: - There's a typo here: "substract" --> subtract.
- Names of days should be capitalized.
```suggestion
`thursdays`|Any integer|Number of Thursdays to add to or subtract from the current date
```

> `fridays`|Any integer|Number of fridays to add to or substract from the current date

Reviewer: - There's a typo here: "substract" --> subtract.
- Names of days should be capitalized.
```suggestion
`fridays`|Any integer|Number of Fridays to add to or subtract from the current date
```

> `saturdays`|Any integer|Number of saturdays to add to or substract from the current date

Reviewer: - There's a typo here: "substract" --> subtract.
- Names of days should be capitalized.
```suggestion
`saturdays`|Any integer|Number of Saturdays to add to or subtract from the current date
```

> `sundays`|Any integer|Number of sundays to add to or substract from the current date

Reviewer: - There's a typo here: "substract" --> subtract.
- Names of days should be capitalized.
```suggestion
`sundays`|Any integer|Number of Sundays to add to or subtract from the current date
```
